File: src/main/java/org/kreps/iotdb/compressor/predictors/LastValuePredictor.java
package org.kreps.iotdb.compressor.predictors;

import org.kreps.iotdb.compressor.Predictor;

public class LastValuePredictor implements Predictor {
    private long storedVal = 0;

    public LastValuePredictor() {}

    public void update(long value) {
        this.storedVal = value;
    }

    public long predict() {
        return storedVal;
    }
}

File: client/src/decompressor/LastValuePredictor.js
import Long from 'long';

class LastValuePredictor {
    constructor() {
        this.storedVal = Long.ZERO;
    }

    update(value) {
        this.storedVal = value;
    }

    predict() {
        return this.storedVal;
    }
}   

export default LastValuePredictor;